geogfn: use bounding box intersection fast paths for Geography

otan opened the pull request:
There are shortcuts in the distance algorithm which involving whether
the bounding boxes intersects. Since we have that for Geography, include
the logic.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52010)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
> There are shortcuts in the distance algorithm which involving whether
> the bounding boxes intersects.

nit: "... which involving ..." needs a grammar fix

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>   

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52010#-:-MDR5juS8-hlNaK_H_3d:b-689r4i)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan committed:
geogfn: use bounding box intersection fast paths for Geography

There are shortcuts in the distance algorithm which involve knowing whether
the bounding boxes intersects. Since we have that for Geography, include
the logic.

Release note: None

otan commented:
bors r=sumeerbhola
